tests/integration/workspace_access/test_redash.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from datetime import timedelta
+
+from databricks.sdk.errors import NotFound
+from databricks.sdk.retries import retried
+from databricks.sdk.service import sql
+
+from databricks.labs.ucx.workspace_access import redash
+from databricks.labs.ucx.workspace_access.groups import GroupManager
+from databricks.labs.ucx.workspace_access.manager import PermissionManager
+from databricks.labs.ucx.workspace_access.redash import RedashPermissionsSupport
+
+logger = logging.getLogger(__name__)
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=13))
+def test_redash_replace_workspace_groups_with_account_groups(
+    ws,
+    sql_backend,
+    inventory_schema,
+    make_ucx_group,
+    make_user,
+    make_query,
+    make_query_policy_permissions,
+):
+    """
+
+    Args:
+        sql_backend (object):
+    """
+    ws_group, _ = make_ucx_group()
+    user = make_user()
+
+    query = make_query()
+    make_query_policy_permissions(
+        object_id=query.id,
+        permission_level=sql.PermissionLevel.CAN_RUN,
+        group_name=ws_group.display_name,
+        user_name=user.display_name,
+    )
+    logger.info(f"Query: {ws.config.host}/sql/editor/{query.id}")
+
+    group_manager = GroupManager(sql_backend, ws, inventory_schema, [ws_group.display_name], "ucx-temp-")
+
+    redash_permissions = RedashPermissionsSupport(
+        ws,
+        [redash.Listing(ws.queries.list, sql.ObjectTypePlural.QUERIES)],
+    )
+
+    permission_manager = PermissionManager(sql_backend, inventory_schema, [redash_permissions])
+
+    permission_manager.inventorize_permissions()
+    state = group_manager.get_migration_state()
```

Comment:
this is too slow and too flaky, rewrite with apply tasks:

```
apply_tasks(
        redash_permissions,
        [
            MigratedGroup.partial_info(group_a, group_b),
        ],
    )
```

see other tests in the folder using the `apply_tasks`

Reply:
This approach would crawl and then apply permissions. But we need to rename the groups in between. So it won't work.

Reply:
updated